### PR TITLE
add try/catch around scheduler to reset sleep state

### LINKED ISSUE #54700: ^C inside pkg-add prompt causes assertion abort
With assertions enabled:
```
julia> using Catlab

       # Initialize workflow object
 │ Package Catlab not found, but a package named Catlab is available from a registry.
 │ Install package?
 │   (@v1.12) pkg> add Catlab
 └ (y/n/o) [y]: ^C
julia: /home/keno/julia/src/scheduler.c:449: ijl_task_get_next: Assertion `jl_atomic_load_relaxed(&ptls->sleep_check_state) == not_sleeping' failed.

[2041131] signal 6 (-6): Aborted
in expression starting at REPL[2]:1
pthread_kill at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
raise at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
abort at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
unknown function (ip: 0x76c6804267da)
__assert_fail at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
ijl_task_get_next at /home/keno/julia/src/scheduler.c:449
poptask at ./task.jl:1154
wait at ./task.jl:1163
uv_write at ./stream.jl:1073
unsafe_write at ./stream.jl:1146
unsafe_write at ./io.jl:803 [inlined]
unsafe_write at ./io.jl:801
write at ./io.jl:807 [inlined]
write at ./stream.jl:1184
jfptr_write_47402 at /home/keno/julia/usr/lib/julia/sys-debug.so (unknown line)
_jl_invoke at /home/keno/julia/src/gf.c:2997
ijl_apply_generic at /home/keno/julia/src/gf.c:3193
```

### PR DESCRIPTION
**Pull request by vtjnash**
Fixes #54700

Mostly just an indentation change, so recommend viewing with whitespace hidden

### CODE REVIEW

**Comment by vtjnash** [comment]
I am somewhat uncertain what happens with all of the control flow here (continue, break, return). The analyzers seem okay with it, but do I need to manually insert a `jl_eh_restore_state(__eh_ct, &__eh)` before each of those?

**Comment by Keno** [comment]
I don't know enough about this code, but CI seems unhappy:
```

[25707] signal 11 (2): Segmentation fault: 11
--
  | in expression starting at none:1
  | ijl_eh_restore_state_noexcept at /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-honeycrisp-HL2F7YQ3XH.0/build/default-honeycrisp-HL2F7YQ3XH-0/julialang/julia-master/src/rtutils.c:300 [inlined]
  | ijl_pop_handler_noexcept at /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-honeycrisp-HL2F7YQ3XH.0/build/default-honeycrisp-HL2F7YQ3XH-0/julialang/julia-master/src/rtutils.c:321
  | #wait#733 at ./condition.jl:139
  | Allocations: 13066918 (Pool: 13066786; Big: 132); GC: 69


```

**Comment by vtjnash** [comment]
Good point. I added those calls where I think I could see they may be necessary, and now CI seems okay?

**Comment by vtjnash** [comment]
This still feels wrong though, since `JL_TRY` changes the meaning of `continue` to mean "jump to end", which may already call `jl_eh_restore_state`. It is idempotent currently, but perhaps undesirable to rely on that?

File: src/scheduler.c [hunk 1]
```diff
@@ -456,144 +456,160 @@ JL_DLLEXPORT jl_task_t *jl_task_get_next(jl_value_t *trypoptask, jl_value_t *q,
...
+            volatile int isrunning = 1;
+            JL_TRY {
+                task = get_next_task(trypoptask, q); // note: this should not yield
+                if (ptls != ct->ptls) {
+                    // sigh, a yield was detected, so let's go ahead and handle it anyway by starting over
+                    ptls = ct->ptls;
+                    if (set_not_sleeping(ptls)) {
+                        JL_PROBE_RT_SLEEP_CHECK_TASK_WAKE(ptls);
+                    }
+                    jl_eh_restore_state_noexcept(__eh_ct, &__eh); // leave
```
Comment (Keno) [inline src/scheduler.c:468]: Not a fan of referencing identifiers that were only introduced inside a preprocessor macro. Can we do `JL_LEAVE` or even `JL_LEAVE_RETURN/JL_LEAVE_CONTINUE`?

**Comment by KristofferC** [comment]
This has some merge conflicts merging it into `backports-release-1.11`

**Comment by vtjnash** [comment]
Visual Studio Code editor almost successfully figured out the merge, it just wasn't sure how much of the last block of code it was supposed to duplicate.

**Pull request merged**